app/plugin_commands.go

```diff
@@ -102,16 +102,25 @@ func (ch *Channels) unregisterPluginCommands(pluginID string) {
 	ch.pluginCommands = remaining
 }
 
-func (a *App) PluginCommandsForTeam(teamID string) []*model.Command {
+// CommandsForTeam returns all the plugin and product commands for the given team.
+func (a *App) CommandsForTeam(teamID string) []*model.Command {
```

Comment:
I merged these two functions in one only. 

I'm open to keeping the double is this feels a bit surprising (or if we're worried about a longer lock for pluginCommandsLock). cc @isacikgoz

Reply:
To me, reducing duplication is a better approach. I'm not concerned about the lock unless it brings some perf degradation.